tiledb/sm/group/group.h

```diff
@@ -80,6 +80,15 @@ class Group {
   /** Destructor. */
   ~Group() = default;
 
+  /**
+   * Creates a TileDB group.
+   *
+   * @param resources The context resources.
+   * @param uri The URI of the group to be created.
+   * @return Status
+   */
+  static Status create(ContextResources& resources, const URI& uri);
```

Comment:
This can return `void`; the only statuses it returns are successful ones, and the implementation already uses `throw_if_not_ok`.

Reply:
There's a separate work item to un`Status` all of the migrated APIs after the migrations are complete. It's out of the scope of each individual PR to unStatus at the time of migration, as it adds an unnecessary level of confusion for the reviewer.

Reply:
I've been the reviewer for most of this. Not combining code movement between source files with other changes has made this whole process much smoother.